### PR TITLE
*: add resource group name into information_schema.processlist

### LINKED ISSUE #40724: Add resource_group_name to inforamtion_schema.processlist
## Enhancement

In resource control feature #38825 ,we also need to add resource group name for process-list entries, and polish information_schema.resource_groups.

### PR DESCRIPTION
**Pull request by BornChanger**
### What problem does this PR solve?

Issue Number: close #40724 

Problem Summary:

### What is changed and how it works?
1. Add `resource_group_name` field to `information_schema.processlist`
image
2. Polish `information_schema.resource_groups`
image

3.  Polish some parser code
4. Don't.set resource group for user when resource group not specified
5. Polish error message when bound resource group doesn't exist
6. Fix issue #40854 

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by glorv** [comment]
Since resource control is disabled by default, I feel weird to see the extra column `resouce_group_name`. Shall we only add this column when resource_group is enabled?

**Comment by BornChanger** [comment]
> Since resource control is disabled by default, I feel weird to see the extra column `resouce_group_name`. Shall we only add this column when resource_group is enabled?

As we discussed, we can't support it. And `resource_group_name` showing doesn't hurt if resource control is disabled. Currently, it only occurs by select to `information_schema.resource_groups`, `show processlist` won't print the column yet.

File: domain/domain_sysvars.go [hunk 1]
```diff
@@ -76,7 +76,10 @@ func (do *Domain) setGlobalResourceControl(enable bool) {
 	} else {
 		variable.DisableGlobalResourceControlFunc()
 	}
-	logutil.BgLogger().Info("set resource control", zap.Bool("enable", enable))
+
+	if enable != variable.EnableResourceControl.Load() {
```
Comment (glorv) [inline domain/domain_sysvars.go:80]: This is false negative, because `variable.EnableResourceControl` is set to `enable` before this function is called, so this log will never be printed.

Comment (BornChanger) [reply domain/domain_sysvars.go]: fixed.

File: infoschema/tables.go [hunk 2]
```diff
@@ -1591,10 +1593,14 @@ var tableMemoryUsageOpsHistoryCols = []columnInfo{
 }
 
 var tableResourceGroupsCols = []columnInfo{
-	{name: "GROUP_ID", tp: mysql.TypeLonglong, size: 64, flag: mysql.NotNullFlag},
-	{name: "GROUP_NAME", tp: mysql.TypeVarchar, size: 512, flag: mysql.NotNullFlag},
-	{name: "RU_PER_SECOND", tp: mysql.TypeLonglong, size: 64},
-	// {name: "BURSTABLE", tp: mysql.TypeVarchar, size: 10},
+	{name: "NAME", tp: mysql.TypeVarchar, size: resourcegroup.MaxGroupNameLength, flag: mysql.NotNullFlag},
+	//{name: "MODE", tp: mysql.TypeVarchar, size: 12, flag: mysql.NotNullFlag},
```
Comment (tiancaiamao) [inline infoschema/tables.go:1597]: Why not clean the code if this is not used any longer?

File: util/gpool/spmc/spmcpool.go [hunk 3]
```diff
@@ -38,23 +38,22 @@ import (
 // TF is the type of the context getter. It is used to get a context.
 // if we don't need to use CT/TF, we can define CT as any and TF as NilContext.
 type Pool[T any, U any, C any, CT any, TF pooltask.Context[CT]] struct {
```
Comment (tiancaiamao) [inline util/gpool/spmc/spmcpool.go:40]: Unrelated change?

**Review by tiancaiamao — APPROVED** [review]

File: util/gpool/spmc/spmcpool.go (hunk 3 above)
Comment (BornChanger) [reply util/gpool/spmc/spmcpool.go:40]: yes. side effect of brazel.

File: infoschema/tables.go (hunk 2 above)
Comment (BornChanger) [reply infoschema/tables.go:1597]: We might need to make them back when multiple modes are needed.

File: infoschema/tables.go [hunk 4]
```diff
@@ -1591,10 +1593,14 @@ var tableMemoryUsageOpsHistoryCols = []columnInfo{
...
+	{name: "RU_FILLRATE", tp: mysql.TypeLonglong, size: 21},
```
Comment (glorv) [inline infoschema/tables.go:1598]: Why does the output not include RU_PER_SEC?

Comment (BornChanger) [reply infoschema/tables.go]: fixed.

File: infoschema/tables.go (hunk 2 above)
Comment (nolouch) [reply infoschema/tables.go:1597]: I prefer clean the code first.

**Review by nolouch — COMMENTED** [review]
rest lgtm

File: infoschema/tables.go (hunk 2 above)
Comment (BornChanger) [reply infoschema/tables.go]: done

**Review by nolouch — APPROVED** [review]

**Review by jackysp — APPROVED** [review]
